Remove polkadot companion detection from url anywhere

cecton opened the pull request:
If you don't mind I would like to remove this feature.

I had a PR failing because the build thought a URL I put was a companion PR while it wasn't. Example here: https://github.com/paritytech/substrate/pull/6555

(The feature is not documented so I didn't update the documentation)

Something that should be ignored by the script: https://github.com/paritytech/polkadot/pull/1393

gabreal commented:
it seems to be mentioned [here](https://github.com/paritytech/substrate/pull/6570/files#diff-4e4c3a60db114d1abcf73821b91c9fe7L34):

> in absence of that string it will check if a polkadot
from this pull request. in absence of that string it will check if a polkadot
pr is mentioned and will use the last one instead.

i'm not sure tbh to remove this here. i agree that specifying things explicitly is better, but in the beginning this feature was requested.

cecton commented:
I meant it wasn't in the CONTRIBUTING doc like the 2 others.

> but in the beginning this feature was requested.

It's also error prone imo.

I think the best would be to use the same branch name but on top of that a bot would automatically add the PR in description when it is detected.

cecton commented on .gitlab-ci.yml:
```diff
@@ -391,6 +391,10 @@ check-polkadot-companion-build:
       artifacts:                   false
   script:
     - ./.maintain/gitlab/check_polkadot_companion_build.sh
+  after_script:
+    - cd polkadot && git rev-parse --abbrev-ref HEAD
+    - echo hello
+    - echo world > /dev/stderr
```
```suggestion
  after_script:
    - cd polkadot && git rev-parse --abbrev-ref HEAD
```

cecton commented:
@gabreal I think this is ready to merge. I removed the detection of the URL without prefix and I added a log at the end of the build that shows more easily which branch/pr has been used as companion pr

gabreal commented on .gitlab-ci.yml:
```diff
@@ -391,6 +391,8 @@ check-polkadot-companion-build:
   script:
     - ./.maintain/gitlab/check_polkadot_companion_build.sh
+  after_script:
+    - cd polkadot && git rev-parse --abbrev-ref HEAD
```
why put it as a after_script? the companion being used is already printed from within the script.
if you want it shown in the end  i would prefer to have this statement there, as this will eventually fail depending on what the script does.

cecton replied:
> why put it as a after_script? the companion being used is already printed from within the script.

Yes but it is printed in the beginning and it is not easy to read. When I check a build failure I usually start at the end of the logs to see what is the error. But here the first important information that everyone should see when they open this log is what branch has been used for the build before even reading the error messages.

>  if you want it shown in the end i would prefer to have this statement there, as this will eventually fail depending on what the script does.

I wanted to place it at the end of the build, no matter if it fails or not (since false success are also important). Where do you think I should move it?

gabreal commented:
I would suggest to add this feature to the CONTRIBUTING doc instead as it is also used in the status check and also the bot uses this kind of processing.
for reference gav's statement was:

> i don't really mind about it being any polkadot url as long as it's in the description of the PR

cecton commented:
> i don't really mind about it being any polkadot url as long as it's in the description of the PR

I doubt he meant that [this](https://github.com/paritytech/polkadot/pull/1393) innocent polkadot url is okay :grin:

I think he meant he doesn't care as long as the polkadot url is not somewhere in the comments of a PR. It should be in one place easy to find: the PR description.

> I would suggest to add this feature to the CONTRIBUTING doc

I'm not sure what feature you mean. The display of the branch being used at the end of the build?

gabreal commented:
@gnunicorn @s3krit pls have a look. do you want to only have explicit companion pr mentionings and disable the fallback to an arbitrary pr mentioned in the substrate's pr description?
(i'm asking because i was once asked to explicitly enable this feature, but things can change)

s3krit commented:
I am personally on the side of things changing - I don't think we should assume *any* polkadot PR URL in the description is a companion PR, and thus block CI if the linked PR is failing. The behaviour isn't clear and, if it is not made very clear in the `CONTRIBUTING.md`, could lead to confusion for external contributors.

sjeohp commented:
@gabreal can you clarify which feature is being disabled?  Currently processbot looks for companion URLs explicitly matching something like `companion: <url>` and will _not_ look for identical branch names.

gabreal commented:
@sjeohp sure. previously there were 3 ways to associate a companion with a substrate pr:
1.) explicit `companion pr: ` in pr description
2.) fallback to arbitrary url mentioned in the description of a pr to a pr in polkadot
3.) match identical branch names

from now on there shall only be option 1.)

sjeohp commented:
Perfect, thanks.

cecton commented:
I updated the doc in the script and the check status script @gabreal
